core/src/main/java/org/apache/accumulo/core/security/crypto/AESCBCEncryptionStrategy.java

```diff
@@ -84,18 +105,35 @@ public OutputStream encryptStream(OutputStream outputStream) throws Exception {
   }
 
   @Override
-  public InputStream decryptStream(InputStream inputStream) throws Exception {
+  public InputStream decryptStream(InputStream inputStream) throws EncryptionStrategyException {
     if (!initialized)
-      throw new Exception("AESEncryptionStrategy not initialized.");
-    int bytesRead = inputStream.read(initVector);
+      throw new EncryptionStrategyException("AESEncryptionStrategy not initialized.");
+    int bytesRead;
+    try {
+      bytesRead = inputStream.read(initVector);
+    } catch (IOException e) {
+      throw new EncryptionStrategyException(e);
+    }
     if (bytesRead != IV_LENGTH_IN_BYTES)
-      throw new Exception("Read " + bytesRead + " bytes, not IV length of " + IV_LENGTH_IN_BYTES
-          + " in decryptStream.");
-
-    Cipher cipher = Cipher.getInstance(this.transformation);
-    cipher.init(Cipher.DECRYPT_MODE, skeySpec, new IvParameterSpec(initVector));
+      throw new EncryptionStrategyException("Read " + bytesRead + " bytes, not IV length of "
+          + IV_LENGTH_IN_BYTES + " in decryptStream.");
+
+    Cipher cipher;
+    try {
+      cipher = Cipher.getInstance(this.transformation);
+      cipher.init(Cipher.DECRYPT_MODE, skeySpec, new IvParameterSpec(initVector));
+    } catch (NoSuchAlgorithmException | NoSuchPaddingException | InvalidKeyException
+        | InvalidAlgorithmParameterException e) {
+      throw new EncryptionStrategyException(e);
+    }
 
     CipherInputStream cis = new CipherInputStream(inputStream, cipher);
     return new BlockedInputStream(cis, cipher.getBlockSize(), 1024);
   }
+
+  @Override
+  public void printCryptoInfoToStream(OutputStream outputStream) {
```

Comment:
Just wondering what your motivation is for having this.  I thought you were opposed to this option on the current implementation?

Reply:
The original code was inconsistent with the responsibilities; the crypto module wrote crypto parameters to the stream for WAL files, and the RFile wrote their own. I originally was thinking of moving this functionality into the file writers, but after reviewing this code and thinking about it, having the EncryptionStrategy responsible for this makes the file writers as crypto-agnostic as possible.